src/WebJobs.Script.WebHost/Standby/StandbyManager.cs

```diff
@@ -70,6 +70,7 @@ public async Task SpecializeHostCoreAsync()
 
             await _languageWorkerChannelManager.SpecializeAsync();
             NotifyChange();
+            await _scriptHostManager.DelayUntilHostReady();
```

Comment:
What does this accomplish exactly?

Reply:
If the job host is restarting, we need to wait for it complete otherwise, restart issued for the specialization never kicks off to orphan jobhost that is using the placeholder content,

Reply:
I think rather than a new Delay here, we need to instead fix the WebJobsScriptHostService.RestartHostAsync method to use a lock for thread synchronization (`lock(_restartLock) { ... }`), rather than the Interlocked pattern it's using now, ignoring if a restart is already in progress. This way, an incoming restart will block until the currently executing restart is completed.

Reply:
I agree with the goal to centralize the logic, however, from an implementation perspective, we should be able to return the same task representing the work, as opposed to blocking those calls.

Reply:
Done!